### PR TITLE
PP-13030: Use enums for the role name

### PR DESCRIPTION
**Pull request by oswaldquek**
When creating a new target service (which will only have a Worldpay test account) and is branched off from an original source service, we want to ensure the admin users on the source service are also admins on the new target service. To this end we want to get admin users for the existing service. We can get all users for a service using `GET /v1/api/services/{serviceExternalId}/users`, but we'd like to introduce a query parameter of `role=admin` for example. Doing this has the advantage of transferring less data across the wire, as the User object is quite large. To this end this PR makes the role an enumerated type which will provide validation on the role query parameter.

This PR:
* Introduces a RoleName enum
* Deletes the RoleDbFixture. Roles are [initialised into the database](https://github.com/alphagov/pay-adminusers/blob/master/src/main/resources/config/initdata/roles.csv), so where there are integration tests, there is never a need to create custom roles. We should use the predefined roles.
* Deletes the RoleEntityFixture for the same reason as above.

This PR does not:
* Add a `role` query parameter to `GET /v1/api/services/{serviceExternalId}/users`
* Enum-ify the role description.

### CODE REVIEW

File: src/test/java/uk/gov/pay/adminusers/resources/InviteResourceServiceCompleteIT.java [hunk 1]
```diff
@@ -21,8 +25,14 @@
 import static uk.gov.pay.adminusers.fixtures.UserDbFixture.userDbFixture;
 
 class InviteResourceServiceCompleteIT extends IntegrationTest {
-    public static final String INVITES_RESOURCE_URL = "/v1/api/invites";
 
+    private Role adminRole;
+
+    @BeforeEach
+    public void setUp() {
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/adminusers/resources/InviteResourceServiceCompleteIT.java:32]: `public` is unnecessary here.

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
I’m in favour of this change. Just a few minor things.

File: src/main/java/uk/gov/pay/adminusers/model/RoleName.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    public String getName() {
+        return name;
+    }
+    
+    public static RoleName fromName(String name) {
+        Optional<RoleName> roleName = roleNames.stream().filter(r -> r.getName().equalsIgnoreCase(name)).findFirst();
+        if (roleName.isEmpty()) {
+            throw undefinedRoleException(name);
+        }
+        return roleName.get();
```
[the remark this answers is not in the document]
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/adminusers/model/RoleName.java:39]: Thanks. I'll go with the last suggestion as it's technically the most performant.

**Comment by oswaldquek** [comment]
> I’m in favour of this change. Just a few minor things.

Yay 🥳

**Review by alexbishop1 — APPROVED** [review]

**Pull request merged**